include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -2352,6 +2359,24 @@ ::fill_JxW_values(AlignedVector<VectorizedArray<Number> > &JxW_values) const
 
 
 
+template <int dim, int n_components_, typename Number>
+inline
+VectorizedArray<Number>
+FEEvaluationBase<dim,n_components_,Number>::JxW(const unsigned int q_point) const
+{
+  Assert (this->J_value != 0, ExcNotImplemented());
```

Comment:
I think `ExcNotImplemented` is not the right assert here. You check that `J_value` has been initialized, right?